libs/blocks/aside/aside.js

```diff
@@ -46,32 +47,98 @@ function decorateStaticLinks(el) {
   textLinks.forEach((link) => { link.classList.add('static'); });
 }
 
+function decorateMedia(el) {
+  if (!(el.classList.contains('medium') || el.classList.contains('large'))) return;
+  const allMedia = Array.from(el.querySelectorAll('div > p video, div > p picture'));
+  allMedia.some((media) => {
+    const parentP = media.closest('p');
+    const siblingP = parentP?.nextElementSibling;
+    if (!siblingP || siblingP.nodeName !== 'P') return false;
+    const siblingText = siblingP.textContent;
+    const hasFormats = siblingText.match(FORMAT_REGEX)?.index;
+    if (hasFormats !== 0) return false;
+    const formats = siblingText.split(': ')[1]?.split(/\s+/);
+    if (formats) {
+      const formatClasses = [];
+      formatClasses.push('format');
+      if (formats.length === 3) formatClasses.push(`desktop-${formats[2]}`);
+      if (formats.length >= 2) formatClasses.push(`tablet-${formats[1]}`);
+      formatClasses.push(`mobile-${formats[0]}`);
+      media.closest('div').classList.add(...formatClasses);
+    }
+    siblingP.remove();
+    media.closest('div').insertBefore(media, parentP);
+    parentP.remove();
+    return true;
+  });
+}
+
+function decorateVideo(container) {
+  const link = container.querySelector('a[href*=".mp4"]');
+  if (!link) return;
+  const isNotLooped = link.hash?.includes('autoplay1');
+  const attrs = `playsinline autoplay ${isNotLooped ? '' : 'loop'} muted`;
+  container.innerHTML = `<video preload="metadata" ${attrs}>
+    <source src="${link.href}" type="video/mp4" />
+  </video>`;
+  container.classList.add('has-video');
+}
+
+function decorateBlockBg(block, node) {
```

Comment:
If we copy/paste a function from another block (in this case I think marquee) we ought to consider instead establishing the function to our shared decorate file (with parameters to handle small diffs). This way we aren't creating tech debt and will not have to edit duplicated functionality in multiple places/blocks as related/new features are requested. I believe this issue of breakpoint BGs is also being talked about in [this ticket](https://jira.corp.adobe.com/browse/MWPW-131763), maybe we can address the duplicated functionality there?

Reply:
@Sartxi The decorateBlockBg is already there in decorate file and we have copied from decorate.js and updated it. The decorateBlockBg in marquee had additional capabilities (like handling focal point which were not needed in aside) hence we did not import the same from marquee. Also video backgrounds were currently not requested for other blocks. 
As a different change, should we merge all these changes in one central location in decorate and let the blocks use the required capability with authoring?